### ISSUE TITLE
icon show [index 0] when tab presed

### ISSUE THREAD

**Issue by KAnggara75** [rafalbednarczuk/curved_navigation_bar#120]
https://user-images.githubusercontent.com/24321218/152817130-a87aa5cf-2413-4ba8-ae18-38df22611b53.mp4

**Comment by wcewong** [comment]
I've been facing a similar issue ever since I upgraded to Flutter 2.10.0.

When an active button (e.g. index = 2) is tapped (despite the button already being active), the button's icon will change to the icon at index 0

**Comment by FrancescoPenasa** [comment]
Me too I've got the same problem, and if I click on the focused item it become the icon at index 0

**Comment by FrancescoPenasa** [comment]
I think I've found the solution by declaring the parameter index of my CurverNavigationBar Widget
```
class _myBottomNavBarState extends State<myBottomNavBar> {
  late int index;
  @override
  void initState() {
    index = widget.index;
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return CurvedNavigationBar(
      height: 70.0,
      index: index, // <-- this line
      backgroundColor: Colors.transparent,
      items: <Widget>[
        Icon(
          Icons.home_rounded,
          color: index == 0 ? kColorDark: kColorGrey,
          size: index == 0 ? 30.0 : 20.0,
        ),
        Icon(
          Icons.person_rounded,
          color: index == 1 ? kColorDark: kColorGrey,
          size: index == 1 ? 30.0 : 20.0,
        ),
        Icon(
          Icons.chat_rounded,
          color: index == 2 ? kColorDark: kColorGrey,
          size: index == 2 ? 30.0 : 20.0,
        ),
      ],
      onTap: (index) {
        setState(() {
          this.index != index ? this.index=index : null;
        });
      },
    );
  }
}
```

**Comment by wcewong** [comment]
> I think I've found the solution by declaring the parameter index of my CurverNavigationBar Widget
> 
> ```
> class _myBottomNavBarState extends State<myBottomNavBar> {
>   late int index;
>   @override
>   void initState() {
>     index = widget.index;
>     super.initState();
>   }
> 
>   @override
>   Widget build(BuildContext context) {
>     return CurvedNavigationBar(
>       height: 70.0,
>       index: index, // <-- this line
>       backgroundColor: Colors.transparent,
>       items: <Widget>[
>         Icon(
>           Icons.home_rounded,
>           color: index == 0 ? kColorDark: kColorGrey,
>           size: index == 0 ? 30.0 : 20.0,
>         ),
>         Icon(
>           Icons.person_rounded,
>           color: index == 1 ? kColorDark: kColorGrey,
>           size: index == 1 ? 30.0 : 20.0,
>         ),
>         Icon(
>           Icons.chat_rounded,
>           color: index == 2 ? kColorDark: kColorGrey,
>           size: index == 2 ? 30.0 : 20.0,
>         ),
>       ],
>       onTap: (index) {
>         setState(() {
>           this.index != index ? this.index=index : null;
>         });
>       },
>     );
>   }
> }
> ```

Thanks a lot mate! I've been trying to solve this issue for hours. I can confirm that your solution works

**Comment by rafalbednarczuk** [comment]
The bug fixed in 1.0.3, rollback changes from 1.0.2
